File: frontend/src/metrics.js
export function trackMetric(event, value = 1) {
    fetch("http://localhost:8000/metrics/custom", {
      method: "POST",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify({ event, value })
    }).catch(err => console.error("Metric send failed:", err));
  }
  

File: backend/app/routers/metrics.py
from fastapi import APIRouter, Request
from services.metrics import record_event

router = APIRouter()

@router.post("/metrics/custom")
async def custom_metrics(request: Request):
    body = await request.json()
    record_event(body["event"], body.get("value", 1.0))
    return {"status": "ok"}
